src/Puzzle.java

```diff
@@ -116,7 +119,7 @@ public static Puzzle fromWord(String word, char primaryLetter,
             );
         }
 
-        chars.remove(primaryLetter);
+        chars.remove(Character.valueOf(primaryLetter));
```

Comment:
Won't this try to use the integer value of primaryLetter as an index, rather than the index of the character? chars.indexOf would seem to be the correct function to use.

Reply:
In Java, Character is a wrapper class of char so that it can be used as a generic (in this case in ArrayList<Character>) as generics are casts to Object under the hood. Character.valueOf returns a Character object of the argument. So this code basically says, don't cast the char to int as an index, treat it as an element value in the list to be removed.

Reply:
Also, using indexOf would required the list be searched twice to delete, rather than once with the boxed element.